homeassistant/components/starlink/config_flow.py

```diff
@@ -0,0 +1,77 @@
+"""Config flow for Starlink."""
+from __future__ import annotations
+
+from typing import Any
+
+from starlink_grpc import ChannelContext, GrpcError, get_id
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import (
+    CONF_NAME,
+    CONF_URL,
+    DATA_RATE_BITS_PER_SECOND,
+    DATA_RATE_GIGABITS_PER_SECOND,
+    DATA_RATE_MEGABITS_PER_SECOND,
+)
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import CONFIG_LINK_SPEED_FORMAT, CONFIG_POLL_RATE, DOMAIN
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default="Starlink"): str,
```

Comment:
Either extract the name from the device, or give it a default name. We like to keep the number of options/questions in a configuration setup flow as low as possible.

The suggestion is to remove the name field